File: test/llvm_swift_test_code/linear_constant/global_14.swift
var g = 0

struct X {
  init() {
    g = 1024
  }
}

var v = X()
@main
struct MyMain {

  static func foo (_ x: Int)-> Int {
    var a = x
    a = a + 1
    return a
  }

  static func main() {
    var a = g
    a = foo(a)
  }
}

File: test/llvm_test_code/linear_constant/global_14.cpp
int g = 0;

struct X {
  X() { g = 1024; }
};

int foo(int x) {
  x = x + 1;
  return x;
}

X var;

int main() {
  int a = g;
  a = foo(a);
  return a;
}
